### PR TITLE
Remove all references to the Maven Wrapper

### PR DESCRIPTION
**Pull request by karthi-au**
# Remove all references to the Maven Wrapper

| **Story ID** | **Story Name** |
| ------------ | -------------- |
| #172483248 |  Remove all references to the Maven Wrapper in the projects listed in the story |

***
## Summary of Changes: 
Remove mvn.cmd and .mvn/wrapper

***
## Review readiness
This pull request is considered a _Work In Progress_ unless all of the following boxes are checked.
Items not required for this PR should be removed from the below list.

- [x] A secret scan was run:
    - [ ] on every commit
    - [ ] on every push
- [x] Existing tests of the following types were run:
    - [ ] unit tests
    - [ ] integration tests
- [ ] Application was run to ensure it is still working as expected
- [ ] Examples given to customers still work as expected
- [ ] Code quality gates have been meet:
    - [ ] Code Coverage
    - [ ] Sonar Quality Gate     
- [x] Ensured no breaking changes to the public interface

***
## PR Reviewer's Checklist
Be sure to give yourself enough time to review the PR.  We are looking for good approvals, not rubber stamped approvals.
- [x] Pull down the branch
- [x] Review the code
- [x] Build the code
- [x] Run tests, application and/or examples locally
- [x] Review code's coverage
- [x] Review code's Sonar quality gate

***
## Reviewers for this PR: 
- [ ] @autonomic-tmc/sdk-team
- [ ] @autonomic-tmc/fe-pr-reviewers

### CODE REVIEW

**Comment by integerAndrew** [comment]
> The repository's [README.md](https://github.com/autonomic-tmc/tmc-auth/tree/kt/remove-maven-wrapper) contains references to the Maven wrapper. Please remove them.
> 
> `./mvnw clean install`
> 
> `./mvnw -Pintegration-tests clean verify`
> 
> `(export TMC_CLIENT_ID=your-client-id && export TMC_CLIENT_SECRET=your-client-secret && ./mvnw -Pintegration-tests clean verify)`

Maybe they should just be updated. We could just do `mvn clean install` and etc.

**Comment by AimmeK** [comment]
Oops!  I meant `update` them, not `remove` then.  Sorry for the confusion.

File: examples/maven-example/README.md [hunk 1]
```diff
@@ -28,9 +28,9 @@ The following properties should be added to your environment or this example's [
 
 From the project directory, run the following:
 
-*Linux/Mac*: `./mvnw -s ./settings.xml spring-boot:run` or `./run.sh`
+*Linux/Mac*: `mvn -s ./settings.xml spring-boot:run` or `./run.sh`
```
Comment (AimmeK) [inline examples/maven-example/README.md:31]: The `tmc-auth` examples still contain their wrappers:   
- [Maven Example](https://github.com/autonomic-tmc/tmc-auth/blob/kt/remove-maven-wrapper/examples/maven-example/mvnw)
- [Gradle Example](https://github.com/autonomic-tmc/tmc-auth/blob/kt/remove-maven-wrapper/examples/gradle-example/gradlew)

I spoke about removing the wrappers from the examples with @integerAndrew when the wrapper was removed as part of the transient license story. In the end, **we decided the wrapper should remain in the examples**.  This way if someone only wants to run the examples they do not need Maven or Gradle. (Maven would only be required if they needed to build tmc-auth.)

If you feel like we should remove the Maven wrapper completely from the examples, please discuss it with the team before doing it.  Let's make sure this is what we want to do.

If the vote is **yes** to remove:
-  Remove EVERYTHING related to the wrapper in the Maven example (wrapper files, wrapper directory, wrapper command, AND documentation)
- Remove the wrapper completely from the Gradle example as well

If the answer is **no**, then just revert this change using the below suggestion.
```suggestion
*Linux/Mac*: `./mvnw -s ./settings.xml spring-boot:run` or `./run.sh`
```

File: examples/maven-example/README.md [hunk 2]
```diff
@@ -28,9 +28,9 @@ The following properties should be added to your environment or this example's [
...
 
-*Windows*: `mvnw -s .\settings.xml spring-boot:run` or `run.bat`
+*Windows*: `mvn -s .\settings.xml spring-boot:run` or `run.bat`
```
Comment (AimmeK) [inline examples/maven-example/README.md:33]: - If you really want to remove the wrapper from the examples, address all of the work outlined in the first comment associated with this file.
- If you are okay with leaving the wrapper in with the example, accept the below suggestion.
```suggestion
*Windows*: `mvnw -s .\settings.xml spring-boot:run` or `run.bat`
```

File: examples/maven-example/README.md [hunk 3]
```diff
@@ -69,9 +69,9 @@ A Maven wrapper is included in this project and references a project specific se
 
 By using the maven wrapper, you will be able to build the project without having to install Maven locally.
 
-Linux/Mac: `./mvnw -s ./settings.xml clean install`
+Linux/Mac: `mvn -s ./settings.xml clean install`
```
Comment (AimmeK) [inline examples/maven-example/README.md:72]: - If you really want to remove the wrapper from the examples, address all of the work outlined in the first comment associated with this file.
- If you are okay with leaving the wrapper in with the example, accept the below suggestion.
```suggestion
Linux/Mac: `./mvnw -s ./settings.xml clean install`
```

File: examples/maven-example/README.md [hunk 4]
```diff
@@ -69,9 +69,9 @@ A Maven wrapper is included in this project and references a project specific se
...
 
-Windows: `mvnw -s .\settings.xml clean install`
+Windows: `mvn -s .\settings.xml clean install`
```
Comment (AimmeK) [inline examples/maven-example/README.md:74]: - If you really want to remove the wrapper from the examples, address all of the work outlined in the first comment associated with this file.
- If you are okay with leaving the wrapper in with the example, accept the below suggestion.
```suggestion
Windows: `mvnw -s .\settings.xml clean install`
```

File: examples/maven-example/run.bat [hunk 5]
```diff
@@ -1 +1 @@
-mvnw spring-boot:run
\ No newline at end of file
+mvn spring-boot:run
```
Comment (AimmeK) [inline examples/maven-example/run.bat:1]: - If you really want to remove the wrapper from the examples, address all of the work outlined in the README.md
- If you are okay with leaving the wrapper in with the example, accept the below suggestion.
```suggestion
mvnw spring-boot:run
```

File: examples/maven-example/run.sh [hunk 6]
```diff
@@ -2,7 +2,7 @@
 set -eo errexit
 
 if [ $# -eq 0 ]; then
-  ./mvnw -s ./settings.xml spring-boot:run
+  mvn -s ./settings.xml spring-boot:run
```
Comment (AimmeK) [inline examples/maven-example/run.sh:5]: - If you really want to remove the wrapper from the examples, address all of the work outlined in the README.md
- If you are okay with leaving the wrapper in with the example, accept the below suggestion.
```suggestion
  ./mvnw -s ./settings.xml spring-boot:run
```

File: examples/maven-example/run.sh [hunk 7]
```diff
@@ -2,7 +2,7 @@
...
 else
-  ./mvnw -s ./settings.xml spring-boot:run -Dtmc-auth.version="$1"
+  mvn -s ./settings.xml spring-boot:run -Dtmc-auth.version="$1"
```
Comment (AimmeK) [inline examples/maven-example/run.sh:7]: - If you really want to remove the wrapper from the examples, address all of the work outlined in the README.md
- If you are okay with leaving the wrapper in with the example, accept the below suggestion.
```suggestion
  ./mvnw -s ./settings.xml spring-boot:run -Dtmc-auth.version="$1"
```

File: examples/maven-example/README.md [hunk 8]
```diff
@@ -41,7 +41,7 @@ See [MavenAuthExample.java](src/main/java/com/autonomic/tmc/example/auth/MavenAu
 
 To access Autonomic's open source dependencies, you can add a Maven url to the repositories section of your `settings.xml` file. See the [settings.xml](settings.xml) file for an example.
 
-To use this settings file, it needs to either be placed in your `~/.m2` directory on Mac or `%userprofile%\.m2` on Windows. The settings file can also be invoked directly with the `-s` command line option (eg, `./mvnw -s these-settings.xml clean verify`).
+To use this settings file, it needs to either be placed in your `~/.m2` directory on Mac or `%userprofile%\.m2` on Windows. The settings file can also be invoked directly with the `-s` command line option (eg, `mvn -s these-settings.xml clean verify`).
```
Comment (AimmeK) [inline examples/maven-example/README.md:44]: Whatever decision is made, I think we should be consistent here.

If we go back to using the Maven Wrapper, then this should be updated to:
```suggestion
To use this settings file, it needs to either be placed in your `~/.m2` directory on Mac or `%userprofile%\.m2` on Windows. The settings file can also be invoked directly with the `-s` command line option (eg, `./mvnw -s these-settings.xml clean verify`).
```

**Comment by karthi-au** [comment]
As discussed with team, leaving the maven wrappers in for now. Updated documentation to use ./mvnw instead of mvn.

Story to investigate usage of maven and gradle wrappers: https://www.pivotaltracker.com/story/show/172596720

**Pull request merged**